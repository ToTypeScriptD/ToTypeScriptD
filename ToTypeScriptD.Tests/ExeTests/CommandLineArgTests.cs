using ApprovalTests;
using Xunit;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading.Tasks;
using System.Text.RegularExpressions;

namespace ToTypeScriptD.Tests.ExeTests
{
    public class CommandLineArgTests : ExeTestBase
    {
        [Fact]
        public void ExeShouldGenerateHelpOnEmptyInput()
        {
            var result = Execute("").StdOut;
            result = Regex.Replace(result, "ToTypeScriptD 0.0.(.*)", "ToTypeScriptD 0.0.*");
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateOutputForMultipleWinmdFiles()
        {
            var result = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Networking.winmd");

            (result.StdOut.Length > 100).ShouldBeTrue(result.StdOut.Length + " should be greater than 100");
        }

        [Fact]
        public void ExeDuplicateAssemblyShouldStillOnlyGenerateOne()
        {
            var resultDup = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");
            var resultNonDup = Execute(@"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");

            // TODO: this test needs a good way to leverage the power of approvals diff capability (but it's not yet supported)

            resultNonDup.StdOut.Trim().ShouldEqual(resultDup.StdOut.Trim());
        }

        [Fact]
        public void ExeShouldBeAbleToGenerateSpecialTypes()
        {
            var resultDup = Execute("--specialTypes");

            resultDup.Verify();
        }
    }
}
