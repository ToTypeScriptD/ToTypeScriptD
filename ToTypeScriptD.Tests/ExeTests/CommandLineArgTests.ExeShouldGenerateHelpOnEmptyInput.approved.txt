ToTypeScriptD 0.0.*
Copyright (C) 2013 Jason Jarrett
 
Usage: TypeScriptD.exe [--specialTypes] [<file1.winmd> ...<fileN.winmd>]

  -s, --specialTypes    Writes the ToTypeScriptD special types to standard out

  --help                Display this help screen.


